reflex-dev/reflex
# [REF-2168] Add share options to custom component commands

## Summary
- Use tomlkit to process pyproject.toml, when publishing, ask user if want to interactively update the info (such as source URL, author email)
- Add a `share` flag for the `publish` command. By default after a successful publish, the command ask the user if they want to share information such as demo app link. This information goes to Reflex backend services.
- Add a `share` command so the same share logic can happen independent of the publish.
## Tests
- [x] Tested locally together with https://github.com/reflex-dev/reflex-hosting/pull/240 
## How to test
- Use the reflex in this PR. Use an existing custom component project or follow the docs to create a new one. The actual implementation does not matter. 
- Point to control plane dev by setting
```
export CP_WEB_URL=https://control-plane.dev.reflexcorp.run
export CP_BACKEND_URL=https://rxcp-dev-control-plane.fly.dev
```
- Run `reflex component share` (it also works when not at the top level of the project, user will be prompted to enter name)
- Answer all the questions (feel free to leave out the optional ones)
- Finally expect to see a new entry on the DB (if you have supabase dev access, look at the `customcomponent` table)
- Expect to see a new entry when querying the public endpoint. One way to easily do that is through swagger https://rxcp-dev-control-plane.fly.dev/docs#/default/get_custom_components_gallery_data_custom_components_gallery_get
